### DISCUSSION TITLE
How to get autocomplete working for commands defined in preamble

### DISCUSSION THREAD

**Discussion by brandon-feder** [James-Yu/LaTeX-Workshop#4321] [Q&A]
I have a preamble which I include by defining the `TEXINPUTS` environment variable then using `\input{my-preamble.tex}`. I would like custom commands defined in my preamble to appear in the autocomplete form. How can I set this up?  It appears to me that this is not supported, but I just wanted to confirm. Thanks!

**Comment by jlelong** [comment]
Have you tried adding the path to the directory containing `my-preamble.tex` to [latex-workshop.latex.texDirs](https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-workshoplatextexdirs)`? See https://github.com/James-Yu/LaTeX-Workshop/wiki/Intellisense#commands-starting-with-


However, I would expect it to work out of the the box without fiddling with `latex-workshop.latex.texDirs` once you have compiled your project with `latexmk`. See the part mentioning the use of `.fls` in https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#the-dependencies.
